python/shotgun_model/shotgun_hierarchy_model.py

```diff
@@ -255,6 +267,30 @@ def canFetchMore(self, index):
     ############################################################################
     # protected methods
 
+    def _check_hierarchy_is_supported(self):
+        """
+        Checks the current Shotgun connection to make sure it supports
+        hierarchy queries.
+        """
+
+        # get the current shotgun connection
+        current_engine = sgtk.platform.current_engine()
+        sg_connection = current_engine.shotgun
+        server_caps = sg_connection.server_caps
```

Comment:
minor: I am not sure this `server_caps` is strictly a public member of the API. To me, a fundamentally more pythonic approach would be to try to execute the operation, and look for a `ShotgunUnsupportedOnServer` exception and then gracefully fall back. This means all the version logic would reside inside the SG API and the exception would be the generic mechaism the Shotgun API used to report incompatibilities to calling code. Does that make sense?

What does the API return today if you run this against an old site?